.github/workflows/publish.yml

```diff
@@ -0,0 +1,95 @@
+name: Publish to OQC Private PyPI
+
+on:
+  push:
+    branches:
+      - main
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
```

Comment:
I believe this triggers on any push to main OR any tag push not AND! (we have this problem on qat at the moment).